### PR TITLE
Added condition for Satellite Upgrading to RHEL8

### PR DESCRIPTION
**Pull request by bangelic**
Added conditional information for upgrading Satellite from RHEL 7 to RHEL 8 per SAT-6752. Removed 'No outstanding updates" bullet in Prerequisites. Updated file name and anchor to be consistent.
@evgeni please review at your earliest convenience

Cherry-pick into:

* [X] Foreman 3.3
* [x] Foreman 3.2
* [ ] Foreman 3.1
* For Foreman 3.0 or older, file a separate PR request

### CODE REVIEW

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-el7-to-el8.adoc:1]: Please adjust the file name to match the anchor/title.

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-el7-to-el8.adoc:98]: You need to specify the code block. If you check your output, you see that the attribute is not replaced.

You can use `[options="nowrap", subs="+quotes,verbatim,attributes"]`. I'd suggest you use that for every code block.

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-el7-to-el8.adoc:240]: This is more of a description what's happening and no "step" that a user has to take. Either reword to "Wait until ..." or use this as description for the first point.

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-el7-to-el8.adoc:41]:
```suggestion
. Any changes you make during this time are not included in the backup on the {RHEL} 8 system on the source machine.
```

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
Some suggestions.

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (evgeni) [inline guides/common/modules/proc_upgrading-el7-to-el8.adoc:99]:
```suggestion
# {package-install-project} leapp leapp-repository
```

`install` is part `package-install-project` already

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (evgeni) [inline guides/common/modules/proc_upgrading-el7-to-el8.adoc:38]:
```suggestion
```

You can't pick and choose.

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (evgeni) [inline guides/common/modules/proc_upgrading-el7-to-el8.adoc:237]: why the difference between Foreman/Satellite? The only difference is EL/RHEL in the "After the system reboots" line, which is the only line that should be conditioned.

**Review by evgeni — CHANGES_REQUESTED** [review]
This is not a full review, I only looked in detail at the "Procedure" for LEAPP.

One overall comment: The Foreman/Katello version of this section reads as "Prerequisites" and "Procedure", which is clear and short. However, the Satellite rendering ends up being 
1. Prerequisites
2. Fresh Install Migration
3. In-Place Migration Using LEAPP
4. Procedure

This lacks structure and doesn't guide the user to a result.

IMHO the structure should be (keeping the "wrong" "Fresh Install Migration" title for easier reference):
1. Different upgrade paths and considerations
  1. Fresh Install Migration
  2. In-Place Migration Using LEAPP
2. Fresh Install Migration
  1. Prerequisites
  2. Procedure
3.  In-Place Migration Using LEAPP
  1. Prerequisites
  2. Procedure

Currently, step 2 is completely missing from this guide (which is fine, we're going step by step, but then there is really no need to talk about the different ways to upgrade yet).

I think this PR would benefit from being reduced to "let's make the LEAPP for Satellite steps correct" and then another one would add the backup/restore way and considerations which one is better. (BTW, backup/restore equally works for Foreman/Katello, there is no reason to condition that part out)

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (bangelic) [reply guides/common/modules/proc_upgrading-el7-to-el8.adoc:23]: Added it back in. Not sure why that got removed initially.

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (bangelic) [reply guides/common/modules/proc_upgrading-el7-to-el8.adoc:30]: @evgeni should I just remove this bullet point?

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (bangelic) [reply guides/common/modules/proc_upgrading-el7-to-el8.adoc:31]: @evgeni like the previous bullet, should I remove it?

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (bangelic) [reply guides/common/modules/proc_upgrading-el7-to-el8.adoc:38]: Removed.

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (bangelic) [reply guides/common/modules/proc_upgrading-el7-to-el8.adoc:237]: Changed it.

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (evgeni) [reply guides/common/modules/proc_upgrading-el7-to-el8.adoc:31]: That's what I'd do, yeah. But I also don't know your original intention behind adding it ;-)

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (evgeni) [reply guides/common/modules/proc_upgrading-el7-to-el8.adoc:32]: same answer :)

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (evgeni) [reply guides/common/modules/proc_upgrading-el7-to-el8.adoc:35]: They need to go into the backup/restore aka clone upgrade, which is not part of this PR, so yes.

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (bangelic) [reply guides/common/modules/proc_upgrading-el7-to-el8.adoc:31]: I saw it in Miriam's document and thought it might be useful. I should've left it alone. :-D

File: guides/common/modules/proc_upgrading-el7-to-el8.adoc
Comment (bangelic) [reply guides/common/modules/proc_upgrading-el7-to-el8.adoc:37]: @evgeni I will change it to **Backup and Restore** for now.

Comment (bangelic) [reply guides/common/modules/proc_upgrading-el7-to-el8.adoc:37]: @ehelms If you think there is a better suggestion for the name, feel free to let me know.

**Comment by bangelic** [comment]
@evgeni Let me know if this can be merged. This needs to be merged ASAP. I will update any changes you think are necessary as soon as I can so we can get this merged.

File: guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc
Comment (evgeni) [inline guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc:11]: Now that you dropped this line in the Satellite case, I realize it's also not really necessary here (but obviously not blocking)

File: guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc
Comment (evgeni) [inline guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc:30]: Given RHEL7→RHEL8 is the only path available for Satellite, we could drop this sentence (it doesn't hurt, but it gives also zero new knowledge to the reader).

File: guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc
Comment (evgeni) [inline guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc:47]: The CentOS part needs to be in `ifdef::foreman-el,katello[] …`

File: guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc
Comment (evgeni) [inline guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc:63]:
```suggestion
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
```
Otherwise `{package-install-project}` is not replaced properly.

File: guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc
Comment (evgeni) [inline guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc:68]:
```suggestion
. Install additional OS specific packages (`leapp-data-centos` for CentOS Stream, `leapp-data-rocky` for Rocky Linux, `leapp-data-almalinux` for AlmaLinux). Not required for {RHEL} based installations.
```

The Foreman-on-RHEL case still needs this comment.

File: guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc
Comment (evgeni) [inline guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc:43]: I still think that having these two sections in here right now completely breaks the flow of the document.

See https://github.com/theforeman/foreman-documentation/pull/1314#pullrequestreview-1003980603 for a longer explanation of my thoughts.

**Review by evgeni — CHANGES_REQUESTED** [review]

File: guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc
Comment (bangelic) [reply guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc:43]: @evgeni I defer to your expertise on this and I will remove these sections from the document. :-)

**Comment by bangelic** [comment]
@evgeni I addressed your comments and recommended changes. Should be good to merge if you don't think anything else is necessary.

File: guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc
Comment (maximiliankolb) [inline guides/common/modules/proc_upgrading-project-from-el7-to-el8.adoc:57]:
```suggestion
. Install additional OS specific packages (`leapp-data-almalinux` for AlmaLinux, `leapp-data-centos` for CentOS Stream, or `leapp-data-rocky` for Rocky Linux).
Note that this is not required for {RHEL} based installations.
```

Please stick to one sentence per line.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
Please reword your git commit messages: https://cbea.ms/git-commit/#seven-rules

**Review by evgeni — APPROVED** [review]
I'll adjust the commit message on merge.

**Comment by evgeni** [comment]
201390fd..cd466254  3.2 -> 3.2
   b742f676..8e2013ba  3.3 -> 3.3

**Comment by evgeni** [comment]
@bangelic no it can not, as Foreman does not support leapp upgrades for 3.1 (satellite does)

**Comment by bangelic** [comment]
@evgeni Oh then we're good already. Thanks!

**Comment by evgeni** [comment]
@bangelic actually, we could pick it, and just adjust the ifdefs to only render for satellite?

**Comment by bangelic** [comment]
> @bangelic actually, we could pick it, and just adjust the ifdefs to only render for satellite?

We want that because it is crucial for Sat 6.11 users per Vendula.

**Pull request merged**
